src/libraries/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs

```diff
@@ -159,6 +159,22 @@ public void AddRangeDerivedArray()
             Assert.Equal(new[] { "a", "b" }, builder);
         }
 
+        [Fact]
+        public void AddRangeSpan()
```

Comment:
Perhaps add coverage for cases where the builder is prepopulated with elements? I'd convert this to a theory taking three arguments: builder elements, range elements, and expected output.

Reply:
Created this test data, and apply it for this test method and more existing builder "AddRange" related test methods.